src/services/selenium/platforms/doordash.py

```diff
@@ -0,0 +1,136 @@
+    DoorDash class inheriting from platform. Has functionality specific to DoorDash.
+    """
+    name = "DoorDash"
+    base_url = "https://www.doordash.com/"
+    order_url = "https://www.doordash.com/orders/"
+
+
+    def __init__(self, service, cookies_path):
+        super().__init__(service, cookies_path)
+        self.driver = self.init_driver()
+    
+    def init_driver(self):
+        driver = uc.Chrome(service=self.service, options=options)
+        return driver
+    
+    def quit_driver(self):
+        return super().quit_driver()
+
+    def login(self):
+        """
+        Logging in to DoorDash. This prompts the user to login and then calls on the save cookies method.
+        """
+        self.driver.get(self.base_url)
+
+        try:
+            wait = WebDriverWait(self.driver, constants.SELENIUM_GLOBAL_DRIVER_WAIT_TIME)
+            login_button = wait.until(EC.element_to_be_clickable((By.XPATH, "//span[contains(@class, 'Text-sc-16fu6d-0')]")))
+            login_button.click()
+                
+            print("Please log in to your Doordash account now.")
+
+            login_success = False
+            start_time = time.time()
+
+            while not login_success and (time.time() - start_time) < 300:
+                try:
+                    account_element = self.driver.find_element(By.XPATH, "//input[@aria-label='Store search: begin typing to search for stores available on DoorDash']")
+                    login_success = True
+                except:
+                    time.sleep(1)
+        
+            if login_success:
+                cookies = self.driver.get_cookies()
+                self.save_cookies(cookies)
+                print("Login successful, cookies saved.")
+            else:
+                print("Login timed out after 5 minutes.")
+
+        except Exception as e:
+            print("Error: ", e)
+
+    def save_cookies(self, cookies):
+        """
+        Creating and saving cookie objects to a .pkl file
+        """
+        cookie_objs = []
+        for cookie in cookies:
+            if cookie["domain"] == ".doordash.com":
+                cookie_objs.append(
+                    {
+                        "name": cookie["name"],
+                        "value": cookie["value"],
+                        "domain": cookie["domain"],
+                    }
+                )
+        try:
+            with open(self.cookies_path, "wb") as file:
+                pickle.dump(cookie_objs, file)
+        except Exception as e:
+            print("Error saving cookies to file: ", e)    
+
+    def access_with_cookies(self):
+        """
+        Loading cookies and accessing the order page using them.
+        """
+        self.driver.get(self.order_url)
+
+        try:
+            self.load_cookies(self.driver)
+            self.driver.refresh()
+            time.sleep(5)
+
+        except Exception as e:
+            print("Error: ", e)
+
+"""
+Main for testing
+"""
+# class Main():
```

Comment:
Since this commented code is there for future reference when we do end up initializing objects of this class in the future, maybe we can leave a note here with some instructions that tell us what to do when porting this code to a different centralized location. Nothing too crazy it could just be:

```python
"""
NOTE: (Baani, 2024-07-11) <brief-instructions-to-port-code>
"""
```

Reply:
Fixed